extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/S3StorageConfig.java

```diff
@@ -36,17 +36,34 @@
   @JsonProperty("sse")
   private final ServerSideEncryption serverSideEncryption;
 
+  /**
+   * S3 transfer config.
+   *
+   * @see S3StorageDruidModule#configure
+   */
+  @JsonProperty("transfer")
+  private final S3TransferConfig s3TransferConfig;
+    
   @JsonCreator
   public S3StorageConfig(
-      @JsonProperty("sse") ServerSideEncryption serverSideEncryption
+      @JsonProperty("sse") ServerSideEncryption serverSideEncryption,
+      @JsonProperty("transfer") S3TransferConfig s3TransferConfig
   )
   {
     this.serverSideEncryption = serverSideEncryption == null ? new NoopServerSideEncryption() : serverSideEncryption;
+    this.s3TransferConfig = s3TransferConfig;
```

Comment:
rather than doing null checks everywhere we should initialize the default config here with transfer turned off.

Reply:
looking at it more closely, I think because you already bind `S3TransferConfig` in JsonConfigProvider.bind, you don't want to also define it as a property of `S3StorageConfig`. I think it should be enough to define it as a property here, or bind it in the storage module and have provided via dependency injection, but not both.

Reply:
Thanks for the feedback. Updated S3StorageConfig and unit tests with default S3TransferConfig